plugins/module_utils/job.py

```diff
@@ -122,430 +114,190 @@ def _job_not_found(job_id, owner, job_name, dd_name, ovrr=None):
-
-        cmd = [tmp.name, " ".join(args)]
-        rc, out, err = module.run_command(args=cmd)
-    except Exception:
-        raise
-    return rc, out, err
-
-
-def _get_return_code_num(rc_str):
-    """Parse an integer return code from
-    z/OS job output return code string.
+def _parse_steps(job_str):
+    """Parse the dd section of output to retrieve step-wise CC's
 
-    Arguments:
-        rc_str {str} -- The return code message from z/OS job log (eg. "CC 0000")
+    Args:
+        job_str (str): The content for a given dd.
 
     Returns:
-        Union[int, NoneType] -- Returns integer RC if possible, if not returns NoneType
+        list[dict]: A list of step names listed as "step executed" the related CC.
     """
+    stp = []
+    if "STEP WAS EXECUTED" in job_str:
+        pile = re.findall(r"(.*?)\s-\sSTEP\sWAS\sEXECUTED\s-\s(.*?)\n", job_str)
+        for match in pile:
+            st = {
+                "step_name": match[0].split()[-1],
+                "step_cc": match[1].split()[-1],
+            }
+            stp.append(st)
 
-    rc = None
-    match = re.search(r"\s*CC\s*([0-9]+)", rc_str)
-    if match:
-        rc = int(match.group(1))
-    return rc
-
-
-def _get_return_code_str(rc_str):
-    """Parse an integer return code from
-    z/OS job output return code string.
+    return stp
 
-    Arguments:
-        rc_str {str} -- The return code message from z/OS job log (eg. "CC 0000" or "ABEND")
 
-    Returns:
-        Union[str, NoneType] -- Returns string RC or ABEND code if possible, if not returns NoneType
-    """
-    rc = None
-    match = re.search(
-        r"(?:\s*CC\s*([0-9]+))|(?:ABEND\s*((?:S|U)[0-9]+)|(?:JCL ERROR))", rc_str
-    )
-    if match:
-        rc = match.group(1) or match.group(2)
-    return rc
+def _zget_job_status(job_id="*", owner="*", job_name="*", dd_name=None):
+    if job_id == "*":
+        job_query = None
+    else:
+        job_query = job_id
+
+    # entries = listing(job_query, owner)   1.2.0 has owner paramn, 1.1 does not
+    entries = listing(job_query)
+
+    final_entries = []
+    if entries:
+        for entry in entries:
+            if owner != "*":
+                if owner != entry.owner:
+                    continue
+            if job_name != "*":
+                if job_name != entry.name:
+                    continue
+
+            job = {}
+
+            job["job_id"] = entry.id
+            job["job_name"] = entry.name
+            job["subsystem"] = ""
+            job["system"] = ""
+            job["owner"] = entry.owner
+
+            job["ret_code"] = {}
+            job["ret_code"]["msg"] = entry.status + " " + entry.rc
+            job["ret_code"]["msg_code"] = entry.rc
+
+            job["ret_code"]["code"] = ""
+            if len(entry.rc) > 0:
+                if entry.rc.isdigit():
+                    job["ret_code"]["code"] = int(entry.rc)
+
+            job["ret_code"]["msg_text"] = entry.status
+
```

Comment:
This is merely a compliment in that appears you kept the code in sync with what the original interface defined the ret_code {} dictionary as (no action just a note for others and kuddos to you).

`
- msg: would would be the job error such as:
     eg; "JCL ERROR" | "ABEND(S|U)xxxx" | "CONV ABEND" | "SEC ERROR" | "SYS FAIL"| CANCELED"| "CC xxxx"
- msg_code: would the string value of the job error reduced down to something more consumable
     eg; "00" ("00" | "04" | "08" | "S0C4"
- msg_txt: is meant for supplying users with something helpful, human readable
	 eg; "The target is unreachable"
- code: if we can convert msg_code into an integer value we return it otherwise we return None
    eg; "00" can be converted to type int 00
`

Reply:
Thanks.